wire-ios-request-strategy/Sources/Message Sending/SessionEstablisher.swift

```diff
@@ -0,0 +1,134 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public enum SessionEstablisherError: Error {
+    case failedToGenerateRequest
+    case failedToParseResponse
+}
+
+public class SessionEstablisher {
+
+    public init(
+        httpClient: HttpClient,
+        context: NSManagedObjectContext
+    ) {
+        self.httpClient = httpClient
+        self.managedObjectContext = context
+    }
+
+    private let httpClient: HttpClient
+    private let managedObjectContext: NSManagedObjectContext
```

Comment:
Can we use `context` instead of `managedObjectContext`? I find it helps de-noise the file quite a lot.

Reply:
It would be nice to know the intended context like the `mainContext` or a `backgroundContext`

Reply:
It's always the `syncContext`